02-express-tutorial/controllers/people.js

```diff
@@ -0,0 +1,56 @@
+let { people } = require("../data");
+
+const getPeople = (req, res) => {
+    res.status(200).json(people);
+}
+  
+const addPerson = (req, res) => {
+    const { name } = req.body;
+    if (!name) {
+      return res.status(400).json({ success: false, message: "Please provide a name!" });
+    }
+    const newPerson = { id: people.length + 1, name };
+    people.push(newPerson);
+    res.status(201).json({ success: true, data: newPerson });
+}
+
+const getPersonById = (req, res) => {
+    const id = parseInt(req.params.id, 10);
+    const person = people.find(p => p.id === id);
+
+    if (!person) {
+      return res.status(404).json({ success: false, message: `Person with id ${id} not found!`});
```

Comment:
in case you need a laugh in the http error code space - https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/418